AutomationHelpers/src/UserCodeCollections/FileLibrary.cs

```diff
@@ -152,9 +153,169 @@ public static void WaitForFile(string path, string pattern, int duration, int in
             }
         }
 
+        /// <summary>
+        /// Compares content of two binary files.
+        /// </summary>
+        /// <param name="filePath1">The relative or absolute path of the first file</param>
+        /// <param name="filePath2">The relative or absolute path of the second file</param>
+        [UserCodeMethod]
+        public static void CompareBinaryFiles(string filePath1, string filePath2)
+        {
+            try
+            {
+                filePath1 = GetPathForFile(filePath1);
+                filePath2 = GetPathForFile(filePath2);
+
+                if (!FilesExist(filePath1, filePath2))
+                {
+                    return;
+                }
+
+                var fi1 = new FileInfo(filePath1);
+                var fi2 = new FileInfo(filePath2);
+
+                if (fi1.Length != fi2.Length)
+                {
+                    Report.Failure("Files '" + filePath1 + "' and '" + filePath2 + "' are not equal.");

```

Comment:
Add reason, e.g. "... are not equal since size is different."

Reply:
Done